### PR TITLE
Provide "global" shim for each bundle

### PR DESCRIPTION
**Pull request by TrySound**
I'm currently bundle my production app with rollup and found unexpected
`global` stuff in consumed modules. I think this should be handled in
library side and provide browser friendly bundle. So the only
requirement could be shimming/replacing `NODE_ENV`.

### CODE REVIEW

**Comment by TrySound** [comment]
Here's my current solution

image

File: packages/jss/.size-snapshot.json [hunk 1]
```diff
@@ -10,21 +10,21 @@
     "gzipped": 5827
   },
   "dist/jss.cjs.js": {
-    "bundled": 48623,
-    "minified": 20945,
-    "gzipped": 5971
+    "bundled": 54298,
+    "minified": 24068,
+    "gzipped": 6908
```
Comment (HenriBeck) [inline packages/jss/.size-snapshot.json]: Why is there such a significant increase in size?

Comment (TrySound) [reply packages/jss/.size-snapshot.json]: Ooops, and here's the problem. node-globals unconditionally polyfills everything. Even `process` object which is used by us for NODE_ENV.

Comment (TrySound) [reply packages/jss/.size-snapshot.json]: Added PR
https://github.com/calvinmetcalf/rollup-plugin-node-globals/pull/16

Comment (HenriBeck) [reply packages/jss/.size-snapshot.json]: I mean we only need to shim `process.env.NODE_ENV`, correct? 
And this should always be replaced by the user.

Comment (TrySound) [reply packages/jss/.size-snapshot.json]: We only need to shim `global` but `process.env.NODE_ENV` should be replaced on user side.

Comment (kof) [reply packages/jss/.size-snapshot.json]: shit I didn't see the size increase

Comment (TrySound) [reply packages/jss/.size-snapshot.json]: Seems like umd bundles size is also increased since `nodeGlobals` is used before `replace`.

Comment (TrySound) [reply packages/jss/.size-snapshot.json]: The simplest work around is own ponyfill which we could reuse in modules.

Comment (HenriBeck) [reply packages/jss/.size-snapshot.json]: are we using anything from `global` other than `process.env.NODE_ENV` or `process.env.VERSION`?

Comment (TrySound) [reply packages/jss/.size-snapshot.json]: `global` is used to create unique identifiers and nothing more. `process.env.VERSION` is replaced statically on our side. `process.env.NODE_ENV` should be replaced statically on user side.

**Comment by TrySound** [comment]
Seems like preventing polyfilling `process` object significantly reduced jss+plugins size.

**Comment by kof** [comment]
awesome!

**Comment by TrySound** [comment]
At least umd bundles are smaller

**Pull request merged**